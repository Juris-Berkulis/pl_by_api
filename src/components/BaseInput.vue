<script setup lang="ts">
interface Props {
    inputValue: string,
};

defineProps<Props>();

defineEmits(['update:inputValue']);
</script>

<template>
<input 
    class="border px-2 py-1 text-sm focus:outline-none" 
    :value="inputValue" 
    @input="$emit('update:inputValue', ($event.target as HTMLInputElement).value)" 
    type="text"
>
</template>

<style scoped>
</style>
