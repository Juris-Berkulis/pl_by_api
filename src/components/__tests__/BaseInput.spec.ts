import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import BaseInput from "../BaseInput.vue";

describe('BaseInput', () => {
    const wrapper = mount(BaseInput, {
        props: {
            inputValue: '',
        }
    });

    it('Snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Contains text', () => {
        const wrapper = mount(BaseInput, {
            props: {
                inputValue: 'aaa',
            }
        });
        expect(wrapper.get('input').attributes('value') === 'aaa');
    });
});
