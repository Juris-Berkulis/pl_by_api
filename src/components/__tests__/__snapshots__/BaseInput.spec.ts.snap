// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseInput > Snapshot 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <input
        class="w-full sm:w-1/2 px-2 py-1 border text-sm focus:outline-none"
        type="text"
      />
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "slotsProxy": null,
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "inputValue": "",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": {
            "update:inputValue": null,
          },
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "inputValue": "",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "inputValue": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            [],
          ],
          "provides": {},
          "proxy": {
            "inputValue": "",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {},
          "slots": {},
          "slotsProxy": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": null,
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [],
            "dynamicProps": [
              "value",
            ],
            "el": <input
              class="w-full sm:w-1/2 px-2 py-1 border text-sm focus:outline-none"
              type="text"
            />,
            "key": null,
            "patchFlag": 40,
            "props": {
              "class": "w-full sm:w-1/2 px-2 py-1 border text-sm focus:outline-none",
              "onInput": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 1,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "input",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "C:/My_projects/022_projects_list_by_api/src/components/BaseInput.vue",
            "__name": "BaseInput",
            "components": {},
            "emits": [
              "update:inputValue",
            ],
            "props": {
              "inputValue": {
                "required": true,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <input
          class="w-full sm:w-1/2 px-2 py-1 border text-sm focus:outline-none"
          type="text"
        />,
        "key": null,
        "patchFlag": 0,
        "props": {
          "inputValue": "",
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "C:/My_projects/022_projects_list_by_api/src/components/BaseInput.vue",
          "__name": "BaseInput",
          "components": {},
          "emits": [
            "update:inputValue",
          ],
          "props": {
            "inputValue": {
              "required": true,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <input
          class="w-full sm:w-1/2 px-2 py-1 border text-sm focus:outline-none"
          type="text"
        />,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.4",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <input
    class="w-full sm:w-1/2 px-2 py-1 border text-sm focus:outline-none"
    type="text"
  />,
}
`;
